src/lib/protocols/connections/ConnectionService.ts

```diff
@@ -182,6 +182,15 @@ class ConnectionService extends EventEmitter {
     return this.connectionRepository.findAll();
   }
 
+  async getById(connectionId: string): Promise<ConnectionRecord | null> {
```

Comment:
What if we have a convention to name every such method that can return null `findByXXX`? So, it will be probably just `getAll()` and similar `getConnections()`, `getCredentials()` and `findBy` for other. I can imagine to have just `find` for the `findById` version, because you can read this from context (method params would have an `id` parameter)